src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Compilation.cs

```diff
@@ -81,7 +81,7 @@ public abstract class Compilation : ICompilation
         public virtual MethodIL GetMethodIL(MethodDesc method)
         {
             // Flush the cache when it grows too big
-            if (_methodILCache.Count > 1000)
+            if (_methodILCache.Count > 3000) // This is a magic number found by compiling System.Private.CoreLib.dll that seems to be reasonable.
```

Comment:
This file is specific to NativeAOT? Did you intend to change this one? ("found by compiling System.Private.CoreLib.dll" doesn't sound like NativeAOT)

Reply:
Woops, I wrote that comment and updated the code with the intention of testing it on System.Private.CoreLib, and it seemed to fix a slowdown I was seeing, but clearly it did nothing.    :) I'll revert this one